src/sonic-bgpcfgd/bgpmon/bgpmon.py

```diff
@@ -106,11 +104,10 @@ def flush_pipe(self, data):
         for key, value in data.items():
             if value is None:
                 # delete case
-                self.pipe.delete(key)
+                self.db.delete(self.db.STATE_DB, key)
```

Comment:
>db [](http://example.com/codeflow?start=21&length=2)

Pipeline is more efficient on a batch of similar operations.

Reply:
Not an exact replacement of redis-py pipe, but you may check if `RedisPipeline` satisfy the usage here.

Reply:
The RedisPipeline  not have delete and hmset methods, to use RedisPipeline we need build a new RedisCommand command and run the command with RedisPipeline. that basically rewrite the delete and hmset method in DBConnector:

https://github.com/Azure/sonic-swss-common/blob/15c0f729648ee9f26ac9aea767d5712db34ca298/common/dbconnector.cpp#L862

Reply:
`delete` should be straightforward. We need more discussion on `hmset`

Reply:
I will check if we can improve current hmset without using lua script.

Reply:
I found the hmset here actually using following API which is not lua script based:

template<typename InputIterator>
void DBConnector::hmset(const std::string &key, InputIterator start, InputIterator stop)
{
    RedisCommand shmset;
    shmset.formatHMSET(key, start, stop);
    RedisReply r(this, shmset, REDIS_REPLY_STATUS);
}

So maybe we can finish this PR first, and improve the lua based hmset with another PR.

Reply:
Could you postpone this change to another future PR? we can merge the other files in this PR first.

Reply:
Fixed, change in this file reverted.